src/core/thread/indirect_sender.cpp

```diff
@@ -422,10 +423,28 @@ void IndirectSender::HandleSentFrameToChild(const Mac::TxFrame &aFrame,
 
     VerifyOrExit(mEnabled);
 
+    if (aError == kErrorNone)
+    {
+        Get<Utils::ChildSupervisor>().UpdateOnSend(aChild);
+    }
+
+    // A zero `nextOffsett` indicates that the sent frame is an empty
+    // frame generated by `PrepareFrameForChild()` when there was no
+    // indirect message in the send queue for the child. This can happen
+    // in the (not common) case where the radio platform does not
+    // support the "source address match" feature and always include
+    // "frame pending" flag in acks to data poll frames. In such a case,
+    // `IndirectSender` prepares and sends an empty frame to the child
+    // after it sends a data poll. Here in `HandleSentFrameToChild()` we
+    // exit quickly if we detect the "send done" is for the empty frame
+    // to ensure we do not update any newly added indirect message after
+    // preparing the empty frame.
```

Comment:
```suggestion
    // A zero `nextOffset` indicates that the sent frame is an empty
    // frame generated by `PrepareFrameForChild()` when there was no
    // indirect message in the send queue for the child. This can happen
    // in the (not common) case where the radio platform does not
    // support the "source address match" feature and always includes
    // "frame pending" flag in acks to data poll frames. In such a case,
    // `IndirectSender` prepares and sends an empty frame to the child
    // after it sends a data poll. Here in `HandleSentFrameToChild()` we
    // exit quickly if we detect the "send done" is for the empty frame
    // to ensure we do not update any newly added indirect message after
    // preparing the empty frame.
```